appeals/web/src/server/appeals/appeal-details/application-decision-date/application-decision-date.mapper.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * @typedef {import('../appeal-details.types.js').WebAppeal} Appeal
+ */
+import { appealShortReference } from '#lib/appeals-formatter.js';
+
+/**
+ * @param {Appeal} appealData
+ * @param {import('@pins/appeals.api').Appeals.SingleAppellantCaseResponse} appellantCaseData
+ * @param {{day: string, month: string, year: string, radio: string}} storedSessionData
+ * @returns {PageContent}
+ */
+export const changeApplicationHasDecisionDatePage = (
+	appealData,
+	appellantCaseData,
+	storedSessionData
+) => {
+	const shortAppealReference = appealShortReference(appealData.appealReference);
+
+	/** @type {PageContent} */
+	const pageContent = {
+		title: `Change the application decision date`,
+		backLinkUrl: `/appeals-service/appeal-details/${appealData.appealId}/appellant-case`,
+		preHeading: `Appeal ${shortAppealReference}`,
+		heading: `Was an application decision made?`,
+		pageComponents: [
+			{
+				type: 'radios',
+				parameters: {
+					name: 'application-decision-radio',
+					id: 'application-decision-radio',
+					fieldSet: {
+						legend: {
+							text: `Was an application decision made?`,
+							isPageHeading: false,
+							classes: 'govuk-fieldset__legend--l'
+						}
+					},
+					items: [
+						{
+							value: 'yes',
+							text: 'Yes',
+							checked:
+								storedSessionData?.radio === 'yes' || appellantCaseData.applicationDecisionDate
+						},
+						{
+							value: 'no',
+							text: 'No',
+							checked:
+								storedSessionData?.radio === 'no' || !appellantCaseData.applicationDecisionDate
+						}
+					]
+				}
+			}
+		]
+	};
+
+	return pageContent;
+};
+
+/**
+ * @param {Appeal} appealData
+ * @param {import('@pins/appeals.api').Appeals.SingleAppellantCaseResponse} appellantCaseData
+ * @param {{day: string, month: string, year: string}} storedSessionData
+ * @returns {PageContent}
+ */
+export const changeApplicationSetDecisionDatePage = (
+	appealData,
+	appellantCaseData,
+	storedSessionData
+) => {
+	const shortAppealReference = appealShortReference(appealData.appealReference);
+
+	let day = storedSessionData?.day ?? '';
+	let month = storedSessionData?.month ?? '';
+	let year = storedSessionData?.year ?? '';
+
+	if (appellantCaseData.applicationDecisionDate && !storedSessionData) {
+		const formattedApplicationDecisionDate = new Date(appellantCaseData.applicationDecisionDate);
+
+		day = formattedApplicationDecisionDate.getDate().toString();
```

Comment:
I don't know if we set the time on `applicationDecisionDate` to make this safe, but we might need to make sure we get the date in `Europe/London` as the server would be running in UTC.

Reply:
Should be resolved now?

Reply:
Would `import { formatInTimeZone } from 'date-fns-tz';` be useful here?

Reply:
Yes potentially. We agreed to leave this for now, as its just `YYYY-MM-DD` handling, it should be OK. Going to do a separate audit of date time handling to check.